### DISCUSSION TITLE
[Table Plugin] Add filtered views

### DISCUSSION THREAD

**Discussion by theRealSuperMario** [zim-desktop-wiki/zim-desktop-wiki#2057] [Ideas]
Hi zim community,

I have loved using zim for many years now due to it's simplicity and the fact that it "just works". However, I am getting increasingly tempted to use other tools because they have some features that I like quite a lot. So I thought maybe we could just build these features into zim?

For instance, the tool [notion.so](notion.so) allows to overlay multiple views onto a table, so that one can filter the table by specifiec categories, for instance.
You can find an example in the attached screenshots.

![image](https://user-images.githubusercontent.com/6517465/80791796-1101b780-8b93-11ea-9f0e-b09f773fcb59.png)

![image](https://user-images.githubusercontent.com/6517465/80791902-59b97080-8b93-11ea-9e49-2261de882e9b.png)

I had a quick look into the zim code and thought that it should be possible to build this feature. If someone gives me a small hint how to do it, I think I might give it a try.

Are there any objections? Because otherwise I will just get started and figure it out.

Cheers

**Comment by jaap-karssenberg** [comment]
Have a look at the plugin documentation for defining "inserted objects" -
this allows you to have a plugin insert a widget in the page. Then you can
code in the plugin both the view and the model for the widget.

-- Jaap

**Comment by introt** [comment]
@theRealSuperMario are you planning to start working on this, or should I move this over to discussions?
